src/pages/VenuePage/VenuePage.tsx

```diff
@@ -80,14 +83,21 @@ const VenuePage = () => {
   const venueName = venue && venue.name;
   // Camp and PartyMap needs to be able to modify this
   // Currently does not work with roome
-  useLocationUpdateEffect(
-    user,
-    venueName
-      ? venueName
-      : profile && profile.lastSeenIn
-      ? profile.lastSeenIn
-      : ""
-  );
+  const isVenueRoom = !!venue?.rooms?.filter(
+    (room) => room.title === profile?.room
+  ).length;
+  const profileRoom = profile?.room;
+  const location =
+    venueName === profileRoom
+      ? venueName ?? ""
+      : isVenueRoom
+      ? profileRoom ?? ""
+      : venueName ?? "";
+  useLocationUpdateEffect(user, location);
+
+  if (!profileRoom && user) {
```

Comment:
Seems this is redundant- the location is always updated above?

Reply:
I removed the updateLocationData from the hook, it only updates the time you've spent in there. The problem was that every single open tab was executing the code.

Reply:
Aha I see now, looks good!